src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -433,24 +480,33 @@ object RuleUtils {
       indexSchema: StructType,
       originalPlan: LogicalPlan,
       filesAppended: Seq[Path]): LogicalPlan = {
+
+    def getFileIndexLocation(plan: LogicalPlan, files: Seq[Path]) = {
+      // Set "basePath" so that partitioned columns are also included in the output schema.
+      val opts = plan match {
+        case _ @ LogicalRelation(
+              _ @ HadoopFsRelation(location: FileIndex, _, _, _, _, _), _, _, _) =>
+          if (location.partitionSchema.isEmpty) {
+            None
+          } else {
+            Some(Hyperspace.getContext(spark).sourceProviderManager.partitionBasePath(plan))
+          }
+        case _: DataSourceV2Relation =>
+          Some(Hyperspace.getContext(spark).sourceProviderManager.partitionBasePath(plan))
+      }
+      val options = opts.flatMap(basePath =>
+        basePath.map(o => Map("basePath" -> o))).getOrElse(Map())
+      new InMemoryFileIndex(spark, files, options, None)
+    }
```

Comment:
Part of `getFileIndexLocation` method's body taken from [L443-L453](https://github.com/microsoft/hyperspace/pull/321/files#diff-8c312528287b51f613809d58a7b14b23c151d65794c099b653605795e0a738edL443-L453).